math/src/fft/serial.rs

```diff
@@ -18,10 +16,10 @@ const MAX_LOOP: usize = 256;
 
 /// Evaluates polynomial `p` in-place over the domain of length `p.len()` in the field specified
 /// by `B` using the FFT algorithm.
-pub fn evaluate_poly<B, E>(p: &mut [E], twiddles: &[B])
+pub fn evaluate_poly<B, I>(p: &mut I, twiddles: &[B])
```

Comment:
Since we can't apply this change to all public functions (i.e., specifically we can't make it work for `evaluate_poly_with_offset()`), we should roll back this and similar changes.

The only two functions which we should change signatures for here are: `permute()` and `fft_in_place()`.

Reply:
Understood, I've rolled back the changes in `serial.rs`. https://github.com/facebook/winterfell/pull/124/commits/da95291284bebe52e7713cd5666b82275b86b173 only introduces the `FftInput` trait in methods `permute` and `fft_in_place`. I was wondering if I should also introduce it similarly in `concurrent.rs` in this PR.